### PR TITLE
Map vote setup checks active players

### PR DESCRIPTION
**Pull request by lessthnthree**
## About The Pull Request
Use the get_active_players proc when starting a map vote to get a more accurate number of active players for map filtering.

This changes the proc where the available maps to choose are determined, changing from counting all players connected (including AFKs) to only counting people who are playing or played in the round.

## Why It's Good For The Game
Lobby screen and nonparticipating ghosts don't count towards map population restrictions. Normal ghosts, alive, dead are included.

This is only applicable for map filtering, there is no change to the voting itself.
## Changelog

:cl: LT3
code: Map vote filtering now counts people playing/played in the round, not all connected clients
fix: Get active players proc now includes unconscious/crit players
/:cl:

### CODE REVIEW

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/datums/votes/map_vote.dm [hunk 1]
```diff
@@ -73,12 +73,14 @@
 		for(var/key in default_choices)
 			choices[key] = 0
 
+	var/eligible_players = get_active_player_count(alive_check = 0, afk_check = 1, human_check = 1)
```
[the remark this answers is not in the document]
Comment (lessthnthree) [reply code/datums/votes/map_vote.dm]: Ah, I was under the impression human_check included the silicons. That should be false, then. The proc does include ghosts who have participated in the round.

**Comment by lessthnthree** [comment]
> This changes map configs to change from counting all players connected (including AFKs) to only counting people who are playing or played in the round

Gonna steal that for the PR.

**Comment by BlueMemesauce** [comment]
why
i guess it makes sense for people in the menu that are probably afk
but observers can still join the next round

**Comment by san7890** [comment]
> why i guess it makes sense for people in the menu that are probably afk but observers can still join the next round

Note it's _roundstart_ observers (allegedly), not people who die mid-shift. I'm still kinda meh on that concept though, but I guess it lessens the potential for hit-and-running votes (not like this is an issue though)

**Comment by lessthnthree** [comment]
Observers (including roundstart) are still able to vote, there is no change to the vote submission process. This change is solely related to the proc that sets up the vote, determining map choices.

**Comment by lessthnthree** [comment]
> > Observers (including roundstart) are still able to vote
> 
> Have you.. read the proc that you're using?
> 
> Roundstart observers are distinctly omitted
> 
> ```
> else if(isobserver(player_mob)) // Ghosts are fine if they were playing once (didn't start as observers)
> 	var/mob/dead/observer/ghost_player = player_mob
> 	if(ghost_player.started_as_observer) // Exclude people who started as observers
> 		continue
> active_players++
> ```
> 
> (Unrelated note but I just noticed the proc has an error and counts people in crit / unconscious as dead)

Yes I've read the proc, again this is only for pre-vote setup (map choices).

`var/datum/map_config/possible_config = config.maplist[map]
		if(possible_config.config_min_users > 0 && active_players < possible_config.config_min_users)
			choices -= map

		else if(possible_config.config_max_users > 0 && GLOB.clients.len > possible_config.config_max_users)
			choices -= map

	return choices
`
The var is only being used to determine if choices are removed as candidates for map vote.

This is unrelated to the vote_submit proc, which is not changing. Observers can still vote.

https://user-images.githubusercontent.com/83487515/220996922-68353b24-4840-4ed9-9d5a-fcf28d2627e1.mp4

**Comment by BlueMemesauce** [comment]
I still don't understand why roundstart observers shouldn't be counted for the next round. Personally, I often observe if its a map I dont enjoy playing as much, then I will join the next round.

**Comment by lessthnthree** [comment]
> I still don't understand why roundstart observers shouldn't be counted for the next round. Personally, I often observe if its a map I dont enjoy playing as much, then I will join the next round.

Maps won't be eliminated from the voting pool due to idle nonparticipating ghosts. Less of a problem with a short AFK timer, but the selection shouldn't be impacted by what your AFK timer settings are in the first place.

**Comment by lessthnthree** [comment]
Not needed with alternate vote

**Pull request closed without merging**